src/apply_conf.rs

```diff
@@ -181,31 +181,35 @@ fn copy_connection_files(
         .to_str()
         .ok_or_else(|| anyhow!("Determining host config path"))?;
 
-    for interface in &host.interfaces {
+    for interface in host.interfaces {
         info!("Processing interface '{}'...", &interface.logical_name);
+        let connections = &interface.connection_ids;
```

Comment:
Can we check the length of this vector and if it's 0 return an error? Or is that going to clash with any of the interface types?

Reply:
We could check this, but I implemented this check in the `generate_conf.rs` `validate_connection_ids()`. So an empty vector should not be possible as output of the generator.

Reply:
In general, yes, but it's more of a sanity check or failure validation where we pretty much guarantee a valid input.

Reply:
Added additional check